### DISCUSSION TITLE
How easy is it to replace MS CodeContracts with MetaLama contracts?

### DISCUSSION THREAD

**Discussion by Orcomp** [metalama/Metalama#348] [Q&A] [answered]
Also, is there an option in MetaLama contracts to disable (or remove) expensive contracts from a release version?

(i.e. We would like some assertions to be permanent but others, the more computationally expensive ones to only be active when we are developing and testing new features, but disabled or removed in release mode.)

**Comment by gfraiteur** [comment, marked as answer]
There is no upgrade path from Microsoft Code Contracts, so I guess it will be manual work.

You can disable precondition, postcondition and invariant checking using the `ArePreconditionsEnabled` , `ArePostconditionsEnabled` and `AreInvariantsEnabled` properties of the `ContractOptions` class. See the [reference documentation](https://doc.postsharp.net/metalama/patterns/contracts/configuring-contracts). 

Specifically:

```c#
public class Fabric : ProjectFabric
{
    public override void AmendProject( IProjectAmender amender )
    {
#if RELEASE
        amender.SetOptions( new ContractOptions {
         ArePreconditionsEnabled = false,
         ArePostconditionsEnabled = false,  
         AreInvariantsEnabled = false  } );
#endif
    }
}
```
